### ISSUE TITLE
Configuration XML File is Missing

### ISSUE THREAD

**Issue by pramodbenda** [viltgroup/jenkins-xcp-ci-plugin#8]
Hi, 

Thanks for providing this Jenkins Plugin for xCP, I tried using it and one thing I noticed is it is not generating the Configuration XML (in the target folder) which is used for xCP Deployment. Do you think I am doing something wrong.

Thanks
Pramod Benda.

**Comment by joaori** [comment]
Hello Pramod,

Are you checking inside the target/ folder of the xCP Application?

In the [wiki](https://github.com/viltgroup/jenkins-xcp-ci-plugin/wiki/build-xcp-application#step-0---prepare-svn-structure) example, it would be **src/TestApplication/target/**

Regards,
João

**Comment by pramodbenda** [comment]
Thanks for your response. I was able to figure out the problem, the issue is with Maven Version (earlier used 3.2.2 as you shown in wiki example), I changed it to 3.0.2 (like it is used in xCP Designer) everything is working fine. But one thing is my project is getting built in target folder not src/<projectName>/target

**Comment by joaori** [comment]
Right, that's probably because you haven't defined any **Local module directory** when setting Source Control for your job, right?

I was able to successfully use the plugin with maven 3.2.2 so it's a bit strange that you are not able to.

**Comment by pramodbenda** [comment]
Yes, you are right I just realized the Local Module Directory is set ".". Still not sure why it is not working for me with Maven 3.2.2.
I have two more questions for you
1. Were you able to build without running xCP Designer?
2. In Deploy xCP Application Instructions Step 0 - Prepare SVN Structure, how can we get the deployment Artifacts into SVN using Jenkins.

Thanks
Pramod Benda

**Comment by joaori** [comment]
Hello Pramod,

I'm afraid you're getting way offtopic here.
1. xCP Designer is mandatory to build a xCP project using maven. As you can see from the pom.xml, there are several plugins dependencies from EMC that depend on it. Please take a look to xCP Designer's help files for more information on how to build an application using maven. (this Jenkins plugins simply tries to facilitate that configuration)
2. Not sure if I understand your question. What do you mean by "deployment artifacts"? You want to know how to upload the WAR generated in the build into SVN?
   I guess you could so some (Windows/Unix) script for it, but I wouldn't recommend you to store binaries into SVN.

**Comment by pramodbenda** [comment]
I am sorry, seems like again I am seeing the same problem that Configuration XML is not getting generated (XIP_xip_2.1.1238_config.xml) is missing, I can see the WAR getting generated. Not sure what should I look at.

**Comment by joaori** [comment]
Was the WAR generated? Please post jenkins **Console Output** of the job execution to see if I can figure it out.

**Comment by pramodbenda** [comment]
Yes, the WAR file is getting generated, I don't the config XML getting generated. I have attached the Job execution Output and Configuration Screenshots for your reference.

Please help me getting this resolved.

[Jenkins Build Output.txt](https://github.com/viltgroup/jenkins-xcp-ci-plugin/files/187896/Jenkins.Build.Output.txt)
[Jenkins_Configuration_Screenshots.docx](https://github.com/viltgroup/jenkins-xcp-ci-plugin/files/187897/Jenkins_Configuration_Screenshots.docx)

**Comment by jahangiranwari** [comment]
@pramodbenda, the **config.xml** file is not generated when building an xCP application from command line using Maven. The xCP Designer 2.1 user guide pg. 353 mentions that

> Maven creates .WAR and .config.xml files for the application created _using xCP Designer_.

Under the hood this plugin uses Maven to build the application from command line.

If your goal is to get a copy of the config.xml then you can use the "Package Application" feature in xCP Designer which will build the WAR and also the config.xml file. You can then save a copy of this config.xml file and use it during deployments.

Additionally, if you are deploying your application across different environments (i.e. Test, Staging, Production) you may want to consider having different config.xml files for each environment.
